### ISSUE TITLE
feat: experiment 05 — backtest comparison (HMM vs buy-and-hold)

### ISSUE THREAD

**Issue by SantiBarrios2002** [SantiBarrios2002/hmm-momentum-trading#15]
## Summary

Implement `experiments/05_backtest_comparison.py` — train HMM on historical data, generate signals on out-of-sample data, backtest the strategy, and compare against buy-and-hold. Partially reproduces Paper Figure 8. Answers: "does the model make money?"

**Standalone:** runs via `python experiments/05_backtest_comparison.py`

### 2. Train/test split
```python
split = int(len(returns) * 0.7)
train_returns = returns[:split]
test_returns = returns[split:]
```
Print the split point, date ranges, and number of observations in each set.

### 3. Train HMM on training data
```python
params, history, _ = baum_welch(train_returns, K=3, max_iter=200,
                                 tol=1e-6, n_restarts=10, random_state=42)
```
Sort states by mu ascending.

### 4. Run inference on test data
```python
predictions, state_probs = run_inference(test_returns, A, pi, mu, sigma2)
```

### 5. Generate signals — two methods
```python
signals_sign = predictions_to_signal(predictions, transfer_fn="sign")
signals_vote = states_to_signal(state_probs, mu)
```

### 6. Backtest all strategies
```python
result_sign = backtest(test_returns, signals_sign, transaction_cost_bps=5)
result_vote = backtest(test_returns, signals_vote, transaction_cost_bps=5)

# Buy-and-hold baseline (no transaction costs — you buy once and hold)
bh_signals = np.ones_like(test_returns)
result_bh = backtest(test_returns, bh_signals, transaction_cost_bps=0)
```

### 7. Print comparison table
```
=== Experiment 05: Backtest Comparison ===
Train: 2015-01-02 to YYYY-MM-DD (XXXX days)
Test:  YYYY-MM-DD to 2024-12-31 (XXX days)

Training 3-state HMM... done.
Running out-of-sample inference... done.

--- Out-of-Sample Performance (5 bps costs for HMM strategies) ---
Strategy            Sharpe   Ann.Return   MaxDrawdown   Turnover
Sign signal          X.XX      XX.X%        XX.X%        X.XX
Weighted vote        X.XX      XX.X%        XX.X%        X.XX
Buy-and-hold         X.XX      XX.X%        XX.X%        0.00
```

### 8. Generate 1 figure

**`figures/05_backtest_comparison.png`**
- 3 cumulative equity curves on the same axes
- All start at 1.0 for comparability
- Lines: "HMM Sign Signal", "HMM Weighted Vote", "Buy-and-Hold"
- x-axis: test period day index (or dates if available)
- y-axis: cumulative value
- Title: "Out-of-Sample Backtest: HMM Strategies vs Buy-and-Hold"
- Legend in upper-left
- Grid enabled

Implementation for the plot:
```python
fig, ax = plt.subplots(figsize=(10, 5))
ax.plot(result_sign["cumulative"], label="HMM Sign Signal", linewidth=1.5)
ax.plot(result_vote["cumulative"], label="HMM Weighted Vote", linewidth=1.5)
ax.plot(result_bh["cumulative"], label="Buy-and-Hold", linewidth=1.5, linestyle="--")
ax.set_title("Out-of-Sample Backtest: HMM Strategies vs Buy-and-Hold")
ax.set_xlabel("Trading Day (test period)")
ax.set_ylabel("Cumulative Value")
ax.legend(loc="upper left")
ax.grid(True, alpha=0.3)
ax.axhline(1.0, color="gray", linestyle=":", alpha=0.5)
fig.tight_layout()
fig.savefig("figures/05_backtest_comparison.png", dpi=150, bbox_inches="tight")
plt.close(fig)
```

## Important Notes

- **The HMM may underperform buy-and-hold on daily SPY data.** The paper uses 1-minute intraday data where momentum effects are stronger. At daily frequency the momentum signal is weak and transaction costs erode returns. This is expected — the experiment demonstrates the full pipeline, not alpha generation.
- **Print an honest interpretation:** if the HMM underperforms, note that this is consistent with the paper's findings that model performance is frequency-dependent.
- **The 1-period signal lag is already handled by `backtest()`** — no manual lagging needed in this script.
- **Sort states before inference** so that "bearish/neutral/bullish" labels are consistent.

## Acceptance Criteria

- [ ] Script runs end-to-end
- [ ] Prints train/test split details
- [ ] Prints formatted comparison table with all 3 strategies
- [ ] Metrics include: Sharpe, annualized return, max drawdown, turnover
- [ ] Figure shows 3 cumulative curves on same axes
- [ ] Figure saved to `figures/05_backtest_comparison.png`
- [ ] Interpretation note printed if HMM underperforms
- [ ] Imports only from `src/`

## Paper Reference
- Paper §7, Figure 8: Sharpe ratios and cumulative returns for Default HMM, PLR HMM, Baum-Welch HMM, IOHMM
- Paper §7: uses 1-period lagged signals for realistic backtesting

## Dependencies
- `src.data.loader.load_daily_prices`
- `src.data.features.log_returns`
- `src.hmm.baum_welch.baum_welch`
- `src.hmm.inference.run_inference`
- `src.strategy.signals.predictions_to_signal`, `states_to_signal`
- `src.strategy.backtest.backtest`
